Plotting Recipe for data on S2 and H2 (#245)
JuliaManifolds/Manifolds.jl: docs/src/manifolds/sphere.md

ArraySphere
```

## Functions on unit spheres
```@autodocs
Modules = [Manifolds]
Pages = ["manifolds/Sphere.jl"]
Order = [:function]
```

## Visualization on `Sphere{2,ℝ}`
You can visualize both points and tangent vectors on the sphere.

!!! note
    There seems to be no unified way to draw spheres in the backends of [Plots.jl](http://docs.juliaplots.org/latest/).
    This recipe currently uses the `seriestype` `wireframe` and `surface`, which does not yet work with the default backend [`GR`](https://github.com/jheinen/GR.jl).

> In general you can plot the surface of the hyperboloid either as wireframe (`wireframe=true`) additionally specifying `wires` (or `wires_lat` and `wires_lon`) for more precision and a `wireframe_color`. The same holds for the plot as a `surface` (which is `false` by default) and its `surface_resolution` (or `surface_resolution_lat` or `surface_resolution_lon`) and a `surface_color`.

Reviewer: Similar comment to the one for hyperboloid applies here.

```@example sphereplot1
using Manifolds, Plots
pyplot()
M = Sphere(2)
pts = [ [1.0, 0.0, 0.0], [0.0, -1.0, 0.0], [0.0, 0.0, 1.0], [1.0, 0.0, 0.0] ]

> scene = plot(M,pts; wireframe_color=colorant"#CCCCCC", markersize=10)

Reviewer: ```suggestion
scene = plot(M, pts; wireframe_color=colorant"#CCCCCC", markersize=10)
```

```

which scatters our points. We can also draw connecting geodesics, which here is a geodesic triangle. Here we discretize each geodesic with 100 points along the geodesic.

> The default value is `geodesic_interpolation=-1` which switches to scatter plot the data.

Reviewer: ```suggestion
The default value is `geodesic_interpolation=-1` which switches to scatter plot of the data.
```

```@example sphereplot1
plot!(scene, M, pts; wireframe=false, geodesic_interpolation=100, linewidth=2)
```

> And we can also add tangent vectors, for example tangents pointing to the middle

Reviewer: ```suggestion
And we can also add tangent vectors, for example tangents pointing towards the geometric center of given points.
```
